### ISSUE TITLE
May I ask some questions?

### ISSUE THREAD

**Issue by yz123456sad** [Visual-AI/PromptCCD#17]
Hello, I noticed the following code in the file promptccd_training_w_gmp.py:

<img width="1292" height="570" alt="Image" src="https://github.com/user-attachments/assets/43a63c1b-fd8f-4dc7-8380-b4be31ec3445" />

This code causes a minor issue. For example, here is the debug information from my log_Kmeans_eval_stage_2.txt:

<img width="692" height="885" alt="Image" src="https://github.com/user-attachments/assets/4fa9952b-9abc-4e0f-b5c7-ca021561cf77" />

Have you noticed that due to catastrophic forgetting, especially between stage 2 and stage 3, the parameters used during model testing are basically those from epoch 0? This means the training of the model has little effect in the subsequent stages. I tried fully training stage 0 and then using only 1 epoch for training in the subsequent stages, and the final accuracy did not drop significantly.
Therefore, I tried, in stage 0, to construct a mapping between the pseudo-labels from your results and the true labels to evaluate the quality of these pseudo-labels. The relevant code is as follows:

<img width="831" height="630" alt="Image" src="https://github.com/user-attachments/assets/04a32fe7-a477-45b4-b738-20c97f605bb3" />

<img width="1220" height="602" alt="Image" src="https://github.com/user-attachments/assets/2f2aebaa-def4-4f9f-a55a-5edc53871076" />

<img width="794" height="642" alt="Image" src="https://github.com/user-attachments/assets/890db7d7-5104-4143-9395-964438a96080" />

<img width="826" height="440" alt="Image" src="https://github.com/user-attachments/assets/6a700157-8bcd-420e-b1d1-e3fc1e1dd038" />

I found that the final accuracy rate is extremely low (of course, it could also be an issue with my debugging code, which is still under investigation). The debugging results are as follows:

<img width="554" height="617" alt="Image" src="https://github.com/user-attachments/assets/162b8d83-3835-4c4b-8a3e-caf2c2e12e1b" />

It can be observed that the accuracy rate is extremely low, which means the accuracy of the prompts given by the model is also extremely low — that is, the prompt module plays a very limited role.

**Comment by fcendra** [comment]
Hi there, thanks for taking interest in our work!

A couple of points that should help:
1. We discussed a very similar observation [here](https://github.com/Visual-AI/PromptCCD/issues/8#issue-2769074420). You can check whether the issue may be caused by the different PyTorch & PyCave versions.

2. On using GMM module outputs as pseudo labels: The GMM in our pipeline is designed as a prompt-selection/generation mechanism — not as a direct classifier. Its role is to guide the model’s representation and prompt routing, not to produce class-aligned pseudo-labels. Because of that, treating the GMM assignments as ground-truth-like pseudo-labels for classification can lead to very low apparent “accuracy,” even when the prompt module is functioning as intended.

Happy to take a closer look and help pinpoint whether this is expected behavior.

Best,
Fernando
